### PR TITLE
Base dux refactor and event dux module

### PR DESCRIPTION
**Pull request by prodhan29**
- refactor the base dux module. mostly renaming the methods and variables
- update readme.md
- event dux module.
- test code for event dux

### CODE REVIEW

**Comment by prodhan29** [comment]
> How does baseDux relate to baseClients?
@moshthepitt 
we have made the baseClient to baseDux module. now there is no baseClient, it is converted to baseDux. which can be used across the application. basic operation like fetch, remove can be use from there.

**Comment by p-netm** [comment]
@prodhan29 Seems like we still have have the baseClients, please remove it.

**Comment by prodhan29** [comment]
> @prodhan29 Seems like we still have have the baseClients, please remove it.

@p-netm Done.

File: clients/core/src/store/ducks/baseDux/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Base Clients Dux Module
```
Comment (moshthepitt) [inline clients/core/src/store/ducks/baseDux/README.md]: I think that this needs to be updated.

File: clients/core/src/store/ducks/baseDux/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+## What am i
+
+Am an abstraction (more like a template), that you can use to create other [dux modules](https://github.com/erikras/ducks-modular-redux).
+
+I think the biggest win with me, is that there is reduced boilerplate when it comes to creating dux modules.
+
+currently , the only opinionated part about me is how i structure the stored data, basically it requires that my action creators are given data as list of objects, where each object bears a property akin to a `primary_key` i.e. this one single field should have unique values that are of type `string| number`.
```
Comment (moshthepitt) [inline clients/core/src/store/ducks/baseDux/README.md]: Lets capitalize the c, i.e. `C`

Comment (prodhan29) [reply clients/core/src/store/ducks/baseDux/README.md]: @moshthepitt  Done !

File: clients/core/src/store/ducks/baseDux/README.md (hunk 1 above)
Comment (prodhan29) [reply clients/core/src/store/ducks/baseDux/README.md]: @moshthepitt Done !

**Pull request merged**